CONTRIBUTING.md

```diff
@@ -44,3 +57,23 @@ project you want to test and updating paths to the target tests in `arguments`.
   </configuration>
 </component>
 ```
+
+## Implementing native interactions
+
+Native methods API is generated from a schema. If you wish to modify any of these methods, follow steps from below:
+
+1. Go to `schema.dart` in the root directory of the repository.
+2. Find a method or a request/response class that you wish to modify, apply your changes.
+3. Run `./gen_from_schema` script to regenerate the contracts.
+
+## Working with patrol_devtools_extension 
+
+If you plan to use the local version of Patrol to test/modify the Patrol DevTools extension, you need to deploy it first. To do this, navigate to the `patrol_devtools_extension` folder and run the `./publish_to_patrol_extension` script.
+
+## Opening a PR
+
+Before opening a PR with your changes, make sure you have updated the changelog for the package you modified.
+
+Use `## Unreleased` title if you're not sure what should be next version number. 
+
+Be aware that `test android emulator` and `test android emulator webview` workflows will fails due to insufficient permission of a contributor. Those will have to be re-run by someone with the write access in the repository.
```

Comment:
I would put this at the beginning of the file, as it applies to every PR. 
You can also add information in the description that mentions issues that have been fixed by this PR. If that problem doesn't exist, it's good practice to create one, or at least describe exactly what the problem was.